components/eamxx/src/physics/nudging/atmosphere_nudging.cpp

```diff
@@ -82,7 +82,7 @@ void Nudging::initialize_impl (const RunType /* run_type */)
   data_in_params.set("Skip_Grid_Checks",true);  // We need to skip grid checks because multiple ranks may want the same column of source data.
   AtmosphereInput data_input0(m_comm,data_in_params);
   data_input0.init(grid_l,host_views,layouts);
-  data_input=data_input0;
+  data_input=std::make_shared<AtmosphereInput>(data_input0);
```

Comment:
This is prob overly complicated. You could just do
```
data_input = std::make_shared<AtmosphereInput>(m_comm,data_in_params);
data_input->init(grid_1,host_views,layouts);
```

Reply:
I have updated, thanks